Shared Framework Code/API/NightscoutAppDataManager.swift

```diff
@@ -49,48 +62,58 @@ public func quickFetch(site: Site, handler: (returnedSite: Site, error: Nightsco
-        quickFetch(site, handler: { (returnedSite, error) -> Void in
-            print("STEP 3:      GET Sever Entries/SGVs for site: \(site.url)")
-            nsAPI.fetchDataForEntries(Constants.EntryCount.NumberForComplication, completetion: { (entries, errorCode) -> Void in
-                site.entries = entries
+    print(">>> Entering \(#function) <<<")
+    print("STARTING fetchSiteData:    Load all available site data for: \(site.url)")
+    let nsAPI = NightscoutAPIClient(url: site.url)
+    var errorToReturn: NightscoutAPIError = .NoError
+    let startDate = NSDate()
+    quickFetch(site, handler: { (returnedSite, error) -> Void in
+        print("\t\tSTEP 3: GET Sever Entries/SGVs ")
+        nsAPI.fetchDataForEntries(Constants.EntryCount.NumberForComplication, completetion: { (entries, errorCode) -> Void in
+            site.entries = entries
+            errorToReturn = errorCode
+            
+            print("\t\t\tSTEP 4: GET Sever CALs/Calibrations")
+            let numberOfCalsNeeded = ((Constants.EntryCount.NumberForComplication * 5) / 60) / 12 + 1
+            nsAPI.fetchCalibrations(numberOfCalsNeeded, completetion: { (calibrations, errorCode) -> Void in
                 errorToReturn = errorCode
                 
-                print("STEP 4:      GET Sever CALs/Calibrations for site: \(site.url)")
-                let numberOfCalsNeeded = ((Constants.EntryCount.NumberForComplication * 5) / 60) / 12 + 1
-                nsAPI.fetchCalibrations(numberOfCalsNeeded, completetion: { (calibrations, errorCode) -> Void in
-                    errorToReturn = errorCode
-                    
-                    guard let calibrations = calibrations else {
-                        return
-                    }
-                    
-                    let cals = calibrations.sort{(item1:Entry, item2:Entry) -> Bool in
-                        item1.date.compare(item2.date) == .OrderedDescending
-                        }.flatMap { $0.cal }
-                    
-                    site.calibrations = cals
-                    
-                    print("STEP 5:      Generate Timeline data for Complication for site: \(site.url)")
-                    let complicationModels = generateComplicationModels(forSite: site, calibrations: site.calibrations)
-                    site.complicationModels = complicationModels
-                    
+                guard let calibrations = calibrations else {
                     NSOperationQueue.mainQueue().addOperationWithBlock({ () -> Void in
-                        print("COMPLETE:    All network operations are complete for site: \(site.url)")
-                        print("DURATION:    The entire process took: \(NSDate().timeIntervalSinceDate(startDate))")
-                        print("STEP 6:      Return Handler to main thread.")
-                        handler(returnedSite: site, error: errorToReturn)
+                        handler(returnedSite: site, error: errorCode)
```

Comment:
when the calibrations come back as a nil, we need to call the handler. @someoneAnyone, this could be one of the places where the UI indicated that the network was hung. If not, it might be good practice to "handle" this type of stuff.